src/openmrs-fetch.ts

```diff
@@ -85,6 +86,17 @@ export function openmrsFetch<T = any>(
        * to help developers understand the problem and debug
        */
 
+      /*
+       *Redirect to given url when redirect on auth failure is enabled
+       */
+      const { redirectAuthFailure } = getConfig("@openmrs/esm-api");
```

Comment:
`getConfig` is an async function that returns a promise. See https://github.com/openmrs/openmrs-esm-module-config/blob/5b4b822f1536ff58885f811164820931cbabb3c8/src/module-config/module-config.ts#L20

We need to make this wait for the promise to resolve. Maybe something like this:

```tsx
return getConfig("@openmrs/esm-api").then(({redirectAuthFailure}) => {
  if (redirectAuthFailure.enabled && redirectAuthFailure.errors.indexOf(response.status) >= 0) {
    navigateToUrl(redirectAuthFailure.url)
  } else {
    return response.text().then(...)
  }
})
```

Reply:
I just realized we haven't published esm-module-config since that time. I will publish a new version to npm now so you can use it.

Reply:
Ah actually wait version 1.0.0 **does** have an async function. So yeah even with esm-module-config@1.0.0 this should be an async function

Reply:
Here it is - https://github.com/openmrs/openmrs-esm-module-config/releases/tag/v1.1.0

Reply:
oops. I missed to see `1.1.0`. Update.